fe/fe-core/src/main/java/com/starrocks/connector/LazyConnector.java

```diff
@@ -0,0 +1,61 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.connector;
+
+import com.starrocks.connector.exception.StarRocksConnectorException;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class LazyConnector implements Connector {
+    private static final Logger LOG = LogManager.getLogger(LazyConnector.class);
+    private final Connector delegate;
+    private final AtomicBoolean inited = new AtomicBoolean(false);
+
+    public LazyConnector(Connector normalConnector) {
+        this.delegate = normalConnector;
+    }
+
+    @Override
+    public ConnectorMetadata getMetadata() {
+        this.init();
+        return delegate.getMetadata();
+    }
+
+    public void init() {
+        synchronized (this) {
+            if (!inited.get()) {
+                try {
+                    delegate.init();
```

Comment:
you can defer create a delegate until here

Reply:
In our initial design,  we required catalog and connector to have a one-to-one correspondence, either succeeding or failing.

Reply:
I don't get how to write that code.